lib/fizzy/execute.cpp

```diff
@@ -357,6 +357,20 @@ inline void convert(OperandStack& stack) noexcept
     stack.top() = static_cast<DstT>(stack.top().as<SrcT>());
 }
 
+/// Performs a bit_cast from SrcT type to DstT type.
+///
+/// This is a formality and should be optimized to empty function in assembly.
+/// Except for f32 -> i32 where pushing the result i32 value to the stack requires zero-extension.
+template <typename SrcT, typename DstT>
+inline void reinterpret(OperandStack& stack) noexcept
+{
+    static_assert(sizeof(SrcT) == sizeof(DstT));
+    const auto src = stack.top().as<SrcT>();
+    DstT dst;
+    __builtin_memcpy(&dst, &src, sizeof(dst));
```

Comment:
Why couldn't `reinterpret_cast` work here instead?

Reply:
1. You cannot do `reinterpret_cast<uint32_t>(f32)`.
2. Using pointers would work: `*reinterpreter_cast<uint32_t*>(&f32)`, but in theory that is not valid use of `reinterpret_cast` and undefined behavior. Using `memcpy` is the safest way of doing type pruning. Although looks ugly.
3. The C++20 there will be `std::bit_cast` - exactly what we need here.